pkg/storegateway/storegatewaypb/rpc.proto

```diff
@@ -0,0 +1,245 @@
+
+  int64 max_resolution_window = 4;
+  repeated Aggr aggregates = 5;
+
+  // Deprecated. Use partial_response_strategy instead.
+  bool partial_response_disabled = 6;
+
+  // TODO(bwplotka): Move Thanos components to use strategy instead. Including QueryAPI.
+  thanos.PartialResponseStrategy partial_response_strategy = 7;
+
+  // skip_chunks controls whether sending chunks or not in series responses.
+  bool skip_chunks = 8;
+
+  // hints is an opaque data structure that can be used to carry additional information.
+  // The content of this field and whether it's supported depends on the
+  // implementation of a specific store.
+  google.protobuf.Any hints = 9;
+
+  // Query step size in milliseconds.
+  // Deprecated: Use query_hints instead.
+  int64 step = 10;
+
+  // Range vector selector range in milliseconds.
+  // Deprecated: Use query_hints instead.
+  int64 range = 11;
+
+  // query_hints are the hints coming from the PromQL engine when
+  // requesting a storage.SeriesSet for a given expression.
+  QueryHints query_hints = 12;
+}
+
+// Analogous to storage.SelectHints.
+message QueryHints {
+  // Query step size in milliseconds.
+  int64 step_millis = 1;
+
+  // The surrounding function or aggregation.
+  Func func = 2;
+
+  // The grouping expression
+  Grouping grouping = 4;
+
+  // Range vector selector.
+  Range range = 5;
+}
+
+message Func {
+  // The function or aggregation name
+  string name = 1;
+}
+
+message Grouping {
+  // Indicate whether it is without or by.
+  bool by = 1;
+
+  // List of label names used in the grouping.
+  repeated string labels = 3;
+}
+
+message Range {
+  int64 millis = 1;
+}
+
+message Stats{
+  int64 index_bytes_fetched = 1;
+}
+
+enum Aggr {
+  RAW = 0;
+  COUNT = 1;
+  SUM = 2;
+  MIN = 3;
+  MAX = 4;
+  COUNTER = 5;
+}
+
+message SeriesResponse {
+  oneof result {
+    /// series contains 1 response series. The series labels are sorted by name.
+    thanos.Series series = 1;
+
+    /// warning is considered an information piece in place of series for warning purposes.
+    /// It is used to warn store API user about suspicious cases or partial response (if enabled).
+    string warning = 2;
+
+    /// hints is an opaque data structure that can be used to carry additional information from
+    /// the store. The content of this field and whether it's supported depends on the
+    /// implementation of a specific store. It's also implementation specific if it's allowed that
+    /// multiple SeriesResponse frames contain hints for a single Series() request and how should they
+    /// be handled in such case (ie. merged vs keep the first/last one).
+    google.protobuf.Any hints = 3;
+
+    Stats stats = 4;
+  }
+}
+
+message LabelNamesRequest {
+  bool partial_response_disabled = 1;
+
+  // TODO(bwplotka): Move Thanos components to use strategy instead. Including QueryAPI.
+  thanos.PartialResponseStrategy partial_response_strategy = 2;
```

Comment:
I think we should take the opportunity to remove all fields we don't use (e.g. this one), remove TODOs, replace "Thanos" with "Mimir".